apps/web/src/components/forms/SignUpForm.tsx

```diff
@@ -0,0 +1,239 @@
+'use client';
+
+import React, { useEffect, useState } from 'react';
+import {
+  TextField,
+  Button,
+  Box,
+  Typography,
+  IconButton,
+  InputAdornment,
+  FormHelperText,
+} from '@mui/material';
+import { Visibility, VisibilityOff } from '@mui/icons-material';
+
+interface SignUpFormData {
+  name: string;
+  email: string;
+  password: string;
+  confirmPassword: string;
+}
+
+const SignUpForm: React.FC = () => {
+  const [formData, setFormData] = useState<SignUpFormData>({
+    name: '',
+    email: '',
+    password: '',
+    confirmPassword: '',
+  });
+
+  const [showPassword, setShowPassword] = useState(false); // State for password visibility
+  const [isButtonDisabled, setIsButtonDisabled] = useState(true); // State for the state of the button
+  const [emailError, setEmailError] = useState(''); // E-mail error state
+  const [passwordError, setPasswordError] = useState(''); // Password error state
+
+  // Email format validation
+  const validateEmail = (email: string) => {
```

Comment:
Because this issue is majorly about building the UI, so it is fine here that we do a email validate, in the future, when we implement the whole logic of login, we should put it into the `/packages/core` lib, so web and native (react native) can reuse the this code.

Reply:
I completely agree that implementing backend validation is the more logical approach. The only reason I wrote some logic for email validation now is to accurately replicate the design from Figma. Removing that logic later will take just a moment, and adjusting it to work with backend values will be quick as well.